transport/src/main/java/io/netty/channel/socket/oio/OioSocketChannel.java

```diff
@@ -143,24 +156,82 @@ protected int doReadBytes(ByteBuf buf) throws Exception {
     }
 
     @Override
-    public ChannelFuture shutdownOutput(final ChannelPromise future) {
+    public ChannelFuture shutdownOutput(final ChannelPromise promise) {
+        EventLoop loop = eventLoop();
+        if (loop.inEventLoop()) {
+            try {
+                socket.shutdownOutput();
+                promise.setSuccess();
+            } catch (Throwable t) {
+                promise.setFailure(t);
+            }
```

Comment:
I'm not sure it is necessary here. It is makes sense where this is currently done in `NioSocketChannel` because it can be called in a few spots ... but here this method just calls itself (so no duplication). The `shutdown` method is different because it must treat the exceptions differently so we can't really share must there.